### PR TITLE
Add xarray Dataset

### PR DESCRIPTION
**Pull request by davidbrochart**

### CODE REVIEW

File: traittypes/traittypes.py [hunk 1]
```diff
@@ -19,6 +19,10 @@ def __getattribute__(self, name):
     import pandas as pd
 except ImportError:
     pd = _DelayedImportError('pandas')
+try:
+    import xarray as xr
+except ImportError:
+    xr = _DelayedImportError('xarray')
```
Comment (vidartf) [inline traittypes/traittypes.py]: This (`xr`) is actually not used anywhere, so you can remove these lines.

Comment (davidbrochart) [reply traittypes/traittypes.py]: Right, and `pd` also (for pandas).

File: traittypes/tests/test_traittypes.py [hunk 2]
```diff
@@ -7,9 +7,10 @@
 from unittest import TestCase
 from traitlets import HasTraits, TraitError, observe, Undefined
 from traitlets.tests.test_traitlets import TraitTestBase
-from traittypes import Array, DataFrame, Series
+from traittypes import Array, DataFrame, Series, Dataset
```
Comment (SylvainCorlay) [inline traittypes/tests/test_traittypes.py]: I would also add DataArray

File: traittypes/traittypes.py [hunk 3]
```diff
@@ -206,3 +202,67 @@ def __init__(self, default_value=Empty, allow_none=False, dtype=None, **kwargs):
...
+
+    def __init__(self, default_value=Empty, allow_none=False, dtype=None, klass=None, **kwargs):
+        if klass is None:
+            klass = self.klass
+        if (klass is not None) and inspect.isclass(klass):
+            self.klass = klass
+        else:
+            raise TraitError('The klass attribute must be a class'
+                                ' not: %r' % klass)
+        self.dtype = dtype
```
Comment (SylvainCorlay) [inline traittypes/traittypes.py]: I think that the `dtype` only makes sense in the `DataArray`  but not in the `Dataset` case.

- Maybe it is a good occasion to also remove it from the `PandasType`, and only keep it in the `Series`.
- Just like `Array` has `dtype = None` at the class level, `Series` and `DataArray` should probably too.

**Comment by SylvainCorlay** [comment]
Thanks for getting started on this!

**Comment by SylvainCorlay** [comment]
This looks good to me! Leaving it open for a bit so that @vidartf can comment further.

**Pull request merged**